torch/nn/_functions/thnn/auto_symbolic.py

```diff
@@ -12,7 +12,17 @@ def leakyrelu_symbolic(g, input, negative_slope, inplace=False):
     return g.op("LeakyRelu", input, alpha_f=negative_slope)
 
 
+def glu_symbolic(g, input, dim):
+    assert input.type().sizes()[dim] % 2 == 0
```

Comment:
Is this a "real assert", or a "we don't support it if this condition doesn't hold" assert? If the latter, replace it with something that throws an exception.

Reply:
It's a real assert https://github.com/colesbury/pytorch/blob/6011d52fc12036b4b244a09e1c9cb226612e640a/torch/lib/THNN/generic/GatedLinearUnit.c#L14